src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2408,6 +2408,37 @@ def import_ssl_cert(cmd, resource_group_name, name, key_vault, key_vault_certifi
                                                 certificate_envelope=kv_cert_def)
 
 
+def sync_kv_cert(cmd, resource_group_name, name, certificate_name, key_vault):
+    Certificate = cmd.get_models('Certificate')
+    client = web_client_factory(cmd.cli_ctx)
+    webapp = client.web_apps.get(resource_group_name, name)
+    if not webapp:
+        raise CLIError("'{}' app doesn't exist in resource group {}".format(name, resource_group_name))
```

Comment:
CLIError is deprecated. @houk-ms to provide more details.

Reply:
Hi @singlessis,  In this case, it's recommended to use [ResourceNotFoundError](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/azclierror.py#L148) for better error categorizing. 

```
from azure.cli.core.azclierror import ResourceNotFoundError

raise ResourceNotFoundError(' ... your error message ...')
```

See the [error handling guidelines](https://github.com/Azure/azure-cli/blob/e2c77ea31f61a32969abf6d2843b88ba038c4548/doc/error_handling_guidelines.md) for more details, we will anounce it soon.